Wine update: 3.20 -> 4.0-rc2

avnik opened the pull request:
###### Motivation for this change

Only 32bit tested (although wine-wow is builds as well)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

avnik committed:
wine{Unstable,Staging}: 3.20 -> 4.0-rc2

Disabling the build of regression tests was proposed in #51341
but this change has been postponed until we reach a concensus.

worldofpeace commented:
cc @yegortimoshenko

yegortimoshenko commented:
@GrahamcOfBorg build wine-staging

worldofpeace reviewed (dismissed):
Can you make the commit message clear that we're updating unstable and staging?

avnik commented:
@worldofpeace done

avnik commented:
@worldofpeace done

avnik commented:
@worldofpeace @yegortimoshenko   updated to friday's 4.0-rc1,  and rebased on fresh master.

ikervagyok commented:
@avnik maybe you could change the title of this pr as well, so people see that 4.0-rc1 is ready ;)

avnik commented:
@ikervagyok  correct, was updated.
I think rc2 would be tomorrow (wine already released, and I waiting for staging (and for my weekend rebuild as well).

@worldofpeace I can't mark your review as resolved by some reasons (probably because I force-update branch)

worldofpeace commented:
@GrahamcOfBorg build wine-staging

worldofpeace commented:
Ahh forgot unstable.
@GrahamcOfBorg build wineUnstable

worldofpeace commented:
Looks good.

@yegortimoshenko Can you check give this a look again so I can merge?

avnik commented:
On Tue, Dec 18, 2018 at 05:24:05PM +0000, worldofpeace wrote:

> Looks good.
> 
> @yegortimoshenko Can you check give this a look again so I can merge?
> 

Or I can drop-in 4.0-rc2 ;)

worldofpeace commented:
> On Tue, Dec 18, 2018 at 05:24:05PM +0000, worldofpeace wrote: Looks good. @yegortimoshenko Can you check give this a look again so I can merge?
> Or I can drop-in 4.0-rc2 ;)

Sure :+1:

avnik commented:
On Tue, Dec 18, 2018 at 09:38:00AM -0800, worldofpeace wrote:

> > On Tue, Dec 18, 2018 at 05:24:05PM +0000, worldofpeace wrote: Looks good. @yegortimoshenko Can you check give this a look again so I can merge?
> > Or I can drop-in 4.0-rc2 ;)
> 
> Sure :+1: 
> 

btw folks, what suggestions for upcomming 4.0...
Put it to "stable" or put it to both?

I want to have wineStaging to evaluate to latest released
version+staging even in rare cases, when stable in one step forward than
unstable.

yegortimoshenko commented on pkgs/misc/emulators/wine/base.nix:
```diff
@@ -7,10 +7,15 @@
+
 stdenv.mkDerivation ((lib.optionalAttrs (! isNull buildScript) {
   builder = buildScript;
 }) // rec {
```
It would be cleaner to remove `rec` here and set `configureFlags = [ "--disable-tests" ] ++ configureFlags`.

worldofpeace commented on pkgs/misc/emulators/wine/base.nix:
```diff
@@ -7,10 +7,15 @@
 with import ./util.nix { inherit lib; };
 
+let configureFlags' = configureFlags ++ [ "--disable-tests" ];
+in
```
I think @avnik said that it saved a larger percentage of build time, which I did notice when having ofborg build this.

lukateras reviewed (approved):
LGTM if https://github.com/NixOS/nixpkgs/pull/51341/commits/8c34f75eefb85eb4552f87328ce1fb5e2549c143 is reverted.

yegortimoshenko commented:
> btw folks, what suggestions for upcomming 4.0... Put it to "stable" or put it to both?

I think the last time (3.0) they actually made development build point to the same one as stable, at the very least on this page: https://web.archive.org/web/20180201114619/https://www.winehq.org/

So just following whatever's on that page should work.

avnik commented:
On Tue, Dec 18, 2018 at 10:02:44AM -0800, Yegor Timoshenko wrote:

> yegortimoshenko commented on this pull request.
> 
> > @@ -7,10 +7,15 @@
>  
>  with import ./util.nix { inherit lib; };
>  
> +let configureFlags' = configureFlags ++ [ "--disable-tests" ];
> +in
> 
> Can we fix the tests?
>

Do we need 10-20 mb of tests in final derivation and extra 20min of
build? Personally I prefer to split them to own output, optional and
make off by default.


> > @@ -7,10 +7,15 @@
>  
>  with import ./util.nix { inherit lib; };
>  
> +let configureFlags' = configureFlags ++ [ "--disable-tests" ];
> +in
> +
>  stdenv.mkDerivation ((lib.optionalAttrs (! isNull buildScript) {
>    builder = buildScript;
>  }) // rec {
> 
> It would be cleaner to remove `rec` here and set `configureFlags = [ "--disable-tests" ] ++ configureFlags`. 
> 

Agreed, will do.

But anyway, I'd like to postpone this specific commit, until further
investigation -- how tests builds/installed and if we can run them
non-interactively during build time, or not.

avnik commented:
On Tue, Dec 18, 2018 at 06:50:43PM +0000, Yegor Timoshenko wrote:

> > btw folks, what suggestions for upcomming 4.0... Put it to "stable" or put it to both?
> 
> I think the last time (3.0) they actually made development build point to the same one as stable, at the very least on this page: https://web.archive.org/web/20180201114619/https://www.winehq.org/
> 
> So just following whatever's on that page should work.
> 

Updated with -rc2 and reverted commit.
